support/cas-server-support-duo-core/src/main/java/org/apereo/cas/adaptors/duo/authn/DuoMfaProviderFactoryBean.java

```diff
@@ -0,0 +1,57 @@
+package org.apereo.cas.adaptors.duo.authn;
+
+import org.apereo.cas.authentication.MultifactorAuthenticationUtils;
+import org.apereo.cas.configuration.CasConfigurationProperties;
+import org.apereo.cas.util.http.HttpClient;
+
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import org.springframework.beans.factory.config.AbstractFactoryBean;
+
+
+/**
+ * Implementation of AbstractFactoryBean used to create Duo provider instances.
+ *
+ * @author Travis Schmidt
+ * @since 5.3.4
+ */
+@Slf4j
+@Setter
+public class DuoMfaProviderFactoryBean extends AbstractFactoryBean<DefaultDuoMultifactorAuthenticationProvider> {
```

Comment:
Do you need te concrete type here as the generic type? Use interface instead?

Reply:
I think interface should work.